lib/Dialect/FIRRTL/Transforms/SpecializeLayers.cpp

```diff
@@ -227,14 +229,14 @@ struct SpecializeLayers {
     // name with the prefix which is then reset to the empty string, and copy it
     // into the new specialize layer reference.
     auto helper = [&](StringAttr ref) -> bool {
-      if (auto specialization = getSpecialization(oldRoot, oldNestedRefs)) {
-        if (*specialization == Specialization::Enable) {
-          // We are enabling this layer, the next non-enabled layer should
-          // include this layer's name as a prefix.
-          prefix.append(ref.getValue());
-          prefix.append("_");
-          return true;
-        }
+      auto specialization = getSpecialization(oldRoot, oldNestedRefs);
+      if (specialization == LayerSpecialization::Enable) {
+        // We are enabling this layer, the next non-enabled layer should
+        // include this layer's name as a prefix.
+        prefix.append(ref.getValue());
+        prefix.append("_");
+        return true;
+      } else if (specialization == LayerSpecialization::Disable) {
```

Comment:
Nit: the `return` makes the `else` for this `if` unnecessary.

Reply:
I wonder why this did not get caught by clang-tidy 🤔.